pkg/server/grpc.go

```diff
@@ -128,3 +133,41 @@ func loadTLSCredentials(certFile, keyFile string) (credentials.TransportCredenti
 
 	return credentials.NewTLS(tlsConfig), nil
 }
+
+func customMetricsInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+	start := time.Now()
+
+	resp, err := handler(ctx, req)
+	if err != nil {
+		slog.Error("error handling request", "error", err)
+	}
+	m := getMetrics()
+	method := info.FullMethod
+	code := status.Code(err).String()
+
+	parts := strings.Split(method, "/")
+	service := ""
+	if len(parts) >= 2 {
+		service = parts[1]
+	}
```

Comment:
I'm not fond of this service parsing code? Especially since it implies we *could* end up with a `""` service string.
Why not just keep the `FullMethod` string? What does trimming the package do for us?